harness_tests/adc/main1.c

```diff
@@ -0,0 +1,74 @@
+#include <test/test.h>
+#include <uart/uart.h>
+#include <adc/adc.h>
+#include <adc/pay.h>
+
+pin_info_t cs = {
+    .port = &ADC_CS_PORT_PAY,
+    .ddr = &ADC_CS_DDR_PAY,
+    .pin = ADC_CS_PIN_PAY
+};
+
+adc_t adc = {
+    .channels = 0x0c00, // poll pins 10 and 11 of the ADC in auto-1 mode
+    .cs = &cs
+};
+
+//any way we can test send_adc?
+
+//test fetch before reset
+void fetch_channel_test(void){
+  //figure out how to test this...
+  fetch_channel(&adc, 11);
+  //choice of 11 based on the adc_test... not sure if this works/is sufficient
+  ASSERT_EQ(adc.mode, MANUAL);
+  //idk what exactly to test regarding adc.channel[c]...
+}
+
+void fetch_all_test(void){
+  fetch_all(&adc);
+  ASSERT_EQ(adc.mode, AUTO1);
+  //what else can i check?
```

Comment:
You could check that the selected auto channels are the only values in the array that got changed.